File: pluginenv/env.go
// Copyright 2026 The Sokel Authors
// SPDX-License-Identifier: Apache-2.0

// Package pluginenv：插件侧环境变量的统一读法。
//
// 只做一件事——把 SOKEL_ 前缀收在一处，别让每个插件各写各的字面量。
//
// 这里**没有**其它前缀的兼容层：认第二个前缀省下的是一次重新部署，
// 换来的是一个没人敢摘的历史包袱——不划算。
package pluginenv

import (
	"os"
	"strings"
)

const prefix = "SOKEL_"

// Get：读 SOKEL_<name>。name 不带前缀，如 Get("TOKEN") / Get("NATS_TOKEN")。
func Get(name string) string { return strings.TrimSpace(os.Getenv(prefix + name)) }

File: sdk-python/sokel/env.py
# Copyright 2026 The Sokel Authors
# SPDX-License-Identifier: Apache-2.0

"""One place to read the plugin's environment variables (mirrors pluginenv on the Go side).

There is no compatibility layer for a second prefix: accepting one saves a single redeploy and buys
a piece of history nobody dares remove.
"""

from __future__ import annotations

import os

PREFIX = "SOKEL_"


def get(name: str) -> str:
    """Read SOKEL_<name>. `name` carries no prefix, e.g. get("TOKEN")."""
    return (os.environ.get(PREFIX + name) or "").strip()


def get_or(name: str, default: str) -> str:
    return get(name) or default
